### PR TITLE
[docs] Update and harmonize guids for deployment

### PR DESCRIPTION
**Pull request by marbre**
This updates and harmonizes the guides for the different deployment configurations. Several identical sections are provided as snippets.

* Updates docs regarding ROCm support
* Adds a subsection on downloading the runtime from a release
* Adds instructions on checking for CUDA and ROCm support

This superseds #18173 and #18655.

skip-ci: Doc updates only

### CODE REVIEW

File: docs/website/docs/guides/deployment-configurations/_iree-compiler-from-release.md
Comment (ScottTodd) [inline docs/website/docs/guides/deployment-configurations/_iree-compiler-from-release.md:1]: > skip-ci: Doc updates only

This shouldn't be needed if you only make changes to the `docs/` directory, FYI.

That pattern should be matched here: https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/build_tools/github_actions/configure_ci.py#L90-L91

which will set the list of enabled jobs to empty: https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/build_tools/github_actions/configure_ci.py#L454-L459

which then causes these conditions to disable the jobs: https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/.github/workflows/ci_linux_x64_clang.yml#L28-L30 https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/.github/workflows/ci_linux_x64_clang_asan.yml#L28-L30
(any job that looks for `skip-ci` will also consider the paths modified)

File: docs/website/docs/guides/deployment-configurations/_iree-run-module-driver-list.md
Comment (ScottTodd) [inline docs/website/docs/guides/deployment-configurations/_iree-run-module-driver-list.md:1]: Are these snippet files published to the website? We may want to exclude them somehow if so.

They would appear on https://iree.dev/sitemap.xml and possibly in search results.

Related: maybe these should go in a `snippets/` subfolder, either at the top level (i.e. `docs/website/docs/snippets/`) or local to where they are used (i.e. `docs/website/docs/guides/deployment-configurations/snippets/`)

**Review by ScottTodd — APPROVED** [review]
Nice!

File: docs/website/docs/guides/deployment-configurations/_iree-compiler-from-release.md
Comment (marbre) [reply docs/website/docs/guides/deployment-configurations/_iree-compiler-from-release.md:1]: Thanks for all the pointers Scott!

File: docs/website/docs/guides/deployment-configurations/_iree-run-module-driver-list.md
Comment (marbre) [reply docs/website/docs/guides/deployment-configurations/_iree-run-module-driver-list.md:1]: They are. As suggested I moved them into a local `snippets/` subfolder (I prefer to keep them close to where they are used) and excluded them via `exclude_docs`.

**Comment by marbre** [comment]
Thanks for the review and the suggestions. All addressed.

**Pull request merged**